src/dataframe/dataframe.jl

```diff
@@ -777,36 +816,58 @@ end
 deletecols!(df::DataFrame, c::Int) = deletecols!(df, [c])
 deletecols!(df::DataFrame, c::Any) = deletecols!(df, index(df)[c])
 
-function deleterows!(df::DataFrame, ind::Union{Integer, UnitRange{Int}})
-    for i in 1:ncol(df)
-        _columns(df)[i] = deleteat!(_columns(df)[i], ind)
-    end
-    df
-end
+"""
+    deleterows!(df::DataFrame, ind)
 
-function deleterows!(df::DataFrame, ind::AbstractVector{Int})
-    ind2 = sort(ind)
-    n = size(df, 1)
+Delete rows from a data frame in place.
 
-    idf = 1
-    iind = 1
-    ikeep = 1
-    keep = Vector{Int}(undef, n-length(ind2))
-    while idf <= n && iind <= length(ind2)
-        1 <= ind2[iind] <= n || error(BoundsError())
-        if idf == ind2[iind]
-            iind += 1
-        else
-            keep[ikeep] = idf
-            ikeep += 1
-        end
-        idf += 1
+### Arguments
+
+* `df` : the DataFrame from which we want to remove rows
+
+* `ind` : a position from which we want to remove rows
+
+### Result
+
+* `::DataFrame` : a `DataFrame` with removed rows. Original will be updated.
+
+### Examples
+
+```jldoctest
+julia> d = DataFrame(a=1:3, b=4:6)
+3×2 DataFrame
+│ Row │ a     │ b     │
+│     │ Int64 │ Int64 │
+├─────┼───────┼───────┤
+│ 1   │ 1     │ 4     │
+│ 2   │ 2     │ 5     │
+│ 3   │ 3     │ 6     │
+
+julia> deleterows!(d, 2)
+2×2 DataFrame
+│ Row │ a     │ b     │
+│     │ Int64 │ Int64 │
+├─────┼───────┼───────┤
+│ 1   │ 1     │ 4     │
+│ 2   │ 3     │ 6     │
+```
+
+"""
+function deleterows!(df::DataFrame, ind)
+    if !isempty(ind) && size(df, 2) == 0
+        throw(BoundsError(summary(df), ind))
```

Comment:
It's weird to pass `summary` here: we're not trying to access the summary string.

Reply:
If the output is unreadable, we should fix the printing of `BoundsError` in Julia. There's currently a special case for `AbstractArray`, but we should probably use `summary` if it's defined for that type, or at least set `limit=true`.

Reply:
I will change it to `ArgumentError` for now. We can use `BoundsError` later as it is not sure if thus change to Base would be backported and when

Reply:
Or to `throw(BoundsError())` - we use similar approach above in the same file. This way we will not change error type in the future when we can print it nicely.